src/core/segment_component.rs

```diff
@@ -22,6 +22,8 @@ pub enum SegmentComponent {
     /// Accessing a document from the store is relatively slow, as it
     /// requires to decompress the entire block it belongs to.
     Store,
+    /// Temporary storage of the documents, before streamed to `Store`.
+    TempStore,
     /// Bitset describing which document of the segment is deleted.
     Delete,
 }
```

Comment:
You need to update `SegmentComponent::iterator()`

Reply:
I missed that, if there other such cases on enums, I think it makes sense to replace them with custom derive macros.

Reply:
I think this is the only case, so macros might be a tad overkill.